### DISCUSSION TITLE
时区的支持 (Time zone support)

### DISCUSSION THREAD

**Discussion by ZhangGuangHuiCoder** [prisma/prisma#20428] [Q&A]
请问该项目有计划对时区做支持吗

Edit: Is there any plan to support time zones in this project?

**Comment by ludralph** [comment]
Hi @zfjhui 👋 

Datetimes are stored in UTC format. UTC will vary according to the timezone you are in. I would recommend setting your server timezone and database timezone to UTC to avoid any confusion and make queries for dates easier.

Is there a specific problem you want addressed about time zone support?

**Reply by ZhangGuangHuiCoder** [reply]
大部分orm和数据库都支持时区配置是有原因的，建议给个配置时区的选项 ，降低使用成本

**Reply by ludralph** [reply]
There is an open issue to improve time zone support. You can add your use case to it and a  :+1: to help our engineers prioritize it.

- https://github.com/prisma/prisma/issues/5051

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ludralph** [reply]
I don't seem to understand your question. Can you please rephrase?

**Reply by ZhangGuangHuiCoder** [reply]
问题主要是，多应用公用数据库，项目的历史应用写入记录时，是以当地时区的值写入的。现在prisma是以utc0写入的，导致不一致。看这个问题已经讨论好久了，不知道prisma团队有没有改进时区支持的计划。🙏@timsuchanek @Jolg42

**Reply by Jolg42** [reply]
@zfjhui could you communicate in English using a translation tool?

Here is the translation of the previous message using Google Translate:

> The main problem is that the multi-application public database, when the historical application of the project is written to the record, is written in the value of the local time zone. Prisma is now written in utc0, causing inconsistencies. Seeing that this issue has been discussed for a long time, I wonder if the prisma team has any plans to improve time zone support.

**Reply by ZhangGuangHuiCoder** [reply]
When multiple applications use  same database, and one of the applications writes data with the local time zone value, if Prisma writes data with UTC+00:00, it may result in inconsistent data storage among different applications. This issue has been discussed for a long time, and we hope that the Prisma team can add a time zone configuration option to better address this issue.

**Reply by HyperClockUp** [reply]
met the same problem. is there any support plan?

**Reply by oyal** [reply]
I have the same needs.

**Reply by firstfu** [reply]
I recently have the same requirements...

**Reply by Jolg42** [reply]
@HyperClockUp check this page for Prisma support policy https://pris.ly/support-policy

**Reply by Jolg42** [reply]
Please take a look at the current timezone issues and check which one relates to your case
https://github.com/prisma/prisma/issues?q=is%3Aopen+label%3A%22topic%3A+timezone+logic+related%22+sort%3Aupdated-desc

If you do not find something that fits your case, please open a new issue 💚

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
